### PR TITLE
DEV: Adds QUnit test

### PR DESCRIPTION
**Pull request by Ahmedgagan**
@jjaffeux

### CODE REVIEW

File: test/javascripts/acceptance/discourse-reactions-double-button-test.js.es6 [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    await visit("/t/-/topic_with_reactions_and_likes");
+
+    assert.ok(
+      exists('[id="post_3"] .discourse-reactions-double-button'),
+      "Has double button"
+    );
+  }
+);
+
+QUnit.test(
```
Comment (jjaffeux) [inline test/javascripts/acceptance/discourse-reactions-double-button-test.js.es6]: Don't think you need QUnit here, do you ?

```suggestion
test(
```

File: test/javascripts/acceptance/discourse-reactions-double-button-test.js.es6 [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  "Doesn't displays double-button when likes and reactions both are present",
```
Comment (jjaffeux) [inline test/javascripts/acceptance/discourse-reactions-double-button-test.js.es6]: This double-button is too much knowledge of internal code... I would prefer a name more related to its usage.

File: test/javascripts/acceptance/discourse-reactions-reaction-button-test.js.es6 [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import { acceptance } from "discourse/tests/helpers/qunit-helpers";
+import { default as ReactionsTopics } from "../fixtures/reactions-topic-fixtues";
+
+acceptance("Discourse reactions reaction button", {
```
Comment (jjaffeux) [inline test/javascripts/acceptance/discourse-reactions-reaction-button-test.js.es6]: Same I want a name more related to the usage

File: test/javascripts/acceptance/likes-and-reactions-test.js.es6 [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+QUnit.test(
+  "Reactions list contains reactions sorted by count",
+  async assert => {
+    await visit("/t/-/topic_with_reactions_and_likes");
+    const expectedSequence =
+      "heartangrylaughingopen_mouthcrythumbsdownnose:t2thumbsup";
+
+    const reactions = find(
+      '[id="post_1"] .discourse-reactions-counter .discourse-reactions-list .reactions .reaction'
+    ).text();
```
Comment (jjaffeux) [inline test/javascripts/acceptance/likes-and-reactions-test.js.es6]: can you try to map/join them to have a nicer and more readable list ?

eg: heart,angry,laughing,...

File: test/javascripts/acceptance/reactions-disabled-test.js.es6 [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    server.get(topicPath, () => helper.response(ReactionsTopics[topicPath]));
+  }
+});
+
+QUnit.test("Doesn't displays reaction-actions", async assert => {
+  await visit("/t/-/topic_with_reactions_and_likes");
+
+  assert.notOk(
+    exists(".discourse-reactions-actions"),
+    "Does not have reaction-actions"
```
Comment (jjaffeux) [inline test/javascripts/acceptance/reactions-disabled-test.js.es6]: same, more usage "Does not show reactions controls" for example

**Comment by jjaffeux** [comment]
Quite good, just look at every textes and try to talk of it as usage and not the name in the code itself

**Comment by Ahmedgagan** [comment]
Sure, ill change the names of the tests 😅

**Comment by Ahmedgagan** [comment]
@jjaffeux Changed the display messages and used MAP/JOIN

**Comment by Ahmedgagan** [comment]
@jjaffeux Added the test for deleted posts.

File: test/javascripts/fixtures/reactions-topic-fixtues.js.es6 [hunk 6]
```diff
@@ -0,0 +1,820 @@
...
...ind all reactions you made.</p>\n<p><img src="//localhost:3000/images/transparent.png" alt="Screenshot 2021-03-08 at 1.34.59 PM" data-orig-src="upload://1NepZH6JY5xZMV3ZqHdisGKwvju" width="1" height="1"></p>\n<h3>See who reacted to your post?</h3>\n<p>In the user page notifications, there is an extra <code>Reactions</code> tab where you will find all reactions made to your posts.</p>\n<p><img src="//localhost:3000/images/transparent.png" alt="Screenshot 2021-03-08 at 1.37.03 PM" data-orig-src="upload://63mpmp96CCpLt3jpGawqlid1XTt" width="1" height="1"></p>\n<h3>Contributors</h3>\n<p><a class="mention" href="/u/awesomerobot">@awesomerobot</a><br>\n<span class="mention">@kris.kotlarek</span><br>\n<span class="mention">@Roman_Rizzi</span></p>\n<h3>Special Thanks</h3>\n<p>Special thanks to <a class="mention" href="/u/awesomerobot">@awesomerobot</a> for being such a nice mentor and for being super helpful in everything <img src="//localhost:3000/images/emoji/twitter/open_hands.png?v=9" title=":open_hands:" class="emoji" alt=":open_hands:"> and <a class="mention-group" href="/groups/team">@team</a> for helping us to make the plugin works perfect by finding bugs as much as possible.</p>',
```
Comment (jjaffeux) [inline test/javascripts/fixtures/reactions-topic-fixtues.js.es6]: Please use a simpler raw, we don't need to have all this text

Comment (Ahmedgagan) [reply test/javascripts/fixtures/reactions-topic-fixtues.js.es6]: Indeed

**Comment by Ahmedgagan** [comment]
@jjaffeux Changed the raw

File: test/javascripts/acceptance/likes-and-reactions-test.js.es6 [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+  await visit("/t/-/topic_with_reactions_and_likes");
+
+  find(
+    '[id="post_1"] .discourse-reactions-counter .discourse-reactions-list .reactions .reaction'
+  ).map((index, currentValue) => {
+    reactions.push(currentValue.innerText);
+  });
+
+  assert.equal(reactions.join("|"), expectedSequence);
```
Comment (Ahmedgagan) [inline test/javascripts/acceptance/likes-and-reactions-test.js.es6]: @jjaffeux This test is failing in the CI, but it always passes in my local. How can I debug it now?

Comment (jjaffeux) [reply test/javascripts/acceptance/likes-and-reactions-test.js.es6]: what is the failure on CI ?

Comment (Ahmedgagan) [reply test/javascripts/acceptance/likes-and-reactions-test.js.es6]:
```
Test Errors
----------------------------------------------
  Test Failed: Reactions list contains reactions sorted by count
    Assertion Failed:
      Expected: heart|angry|laughing|open_mouth|cry|thumbsdown|nose:t2|thumbsup, Actual: |||||||
    at eval (acceptance/likes-and-reactions-test:23:3)
    at obj (discourse-loader:364:31)
    at tryFinally (discourse-loader:188:14)
    at require (discourse-loader:361:5)
    at eval (discourse/tests/setup-tests:229:9)
    at Array.forEach (<anonymous>)
    at setupTestsCommon (discourse/tests/setup-tests:223:36)
```

File: test/javascripts/acceptance/likes-and-reactions-test.js.es6 [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+test("Reactions list contains reactions sorted by count", async (assert) => {
+  const reactions = [];
+  const expectedSequence =
+    "heart|angry|laughing|open_mouth|cry|thumbsdown|nose:t2|thumbsup";
+
+  await visit("/t/-/topic_with_reactions_and_likes");
+
+  find(
+    '[id="post_1"] .discourse-reactions-counter .discourse-reactions-list .reactions .reaction'
```
Comment (jjaffeux) [inline test/javascripts/acceptance/likes-and-reactions-test.js.es6]: why don't we do `#post_1` here ?

Comment (Ahmedgagan) [reply test/javascripts/acceptance/likes-and-reactions-test.js.es6]: Used `#post_1` in the latest commit

**Comment by jjaffeux** [comment]
Can you ensure tests pass here ?

**Comment by Ahmedgagan** [comment]
> Can you ensure tests pass here ?

This test is passing in my local but failing here. I am not getting what's happening

**Comment by jjaffeux** [comment]
oh yes right forgot about this... need to dig this

**Review by jjaffeux — APPROVED** [review]

**Pull request merged**